src/lib.rs

```diff
@@ -387,19 +405,20 @@ impl<'a> Id<'a> {
     ///
     /// Passing an invalid string (containing spaces, brackets,
     /// quotes, ...) will return an empty `Err` value.
-    pub fn new<Name: Into<Cow<'a, str>>>(name: Name) -> Result<Id<'a>, ()> {
+    pub fn new<Name: Into<Cow<'a, str>>>(name: Name) -> Result<Id<'a>, IdError> {
         let name = name.into();
         {
             let mut chars = name.chars();
             match chars.next() {
                 Some(c) if is_letter_or_underscore(c) => {}
-                _ => return Err(()),
+                Some(c) => return Err(IdError::InvalidStartChar(c)),
+                _ => return Err(IdError::EmptyName)
```

Comment:
```suggestion
                None => return Err(IdError::EmptyName)
```
I'd like this to be explicit to avoid accidentally having a catch-all in certain cases. For now this suggestion is somewhat useless. Feel free to not change this.